Automattic/jetpack
# Goodreads Block: Fix Multiple Blocks on Same Page

## Proposed changes:

I was testing the Goodreads block a little more, so just four small changes:

- The block should probably default in descending order (ie. most recent books first) - that's what the widget currently does, and it seems most intuitive. 
- A couple of minor code enhancements - I'm not sure that `displayPreview` is necessary, and that also removes the hardcoded check for the block preview.
- There was an error caused by the block preview being an integer whereas `goodreadsId` elsewhere is a string. It probably seems more fitting to make this an integer. 
- Ensures that the `wp_enqueue_script` handle is unique so that the block works if you insert multiple on one page.

## Jetpack product discussion


N/A.

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:

* Enable beta blocks: `add_filter( 'jetpack_blocks_variation', function () { return 'beta'; } );`
* Add two Goodreads blocks to your site for different profiles
* Confirm that on the front-end, they both now appear rather than only the first one now appearing

Since this changes `displayPreview`, it might also be worth checking that Matt's profile appears in the block preview and that you can successfully edit the URL. 

<img width="671" alt="Screenshot 2024-02-22 at 18 12 56" src="https://github.com/Automattic/jetpack/assets/43215253/474be48e-b55e-4b5f-8b65-dbc23f76d9ef">
